packages/astro/src/core/config.ts

```diff
@@ -266,6 +266,24 @@ interface LoadConfigOptions {
 	flags?: Flags;
 }
 
+/** Resolve the file URL of the user's `astro.config.js|cjs|mjs|ts` file */
+export async function resolveConfigURL(configOptions: LoadConfigOptions): Promise<URL|undefined> {
```

Comment:
refactor loadConfig() to call this internally? seems like a lot of copy-paste would be removed

Reply:
Yes I'd prefer this! This was supposed to just **resolve** the config but I had to rollback that change temporarily. Definitely want to refactor this, though.

Reply:
Decided _not_ to tackle this now, but I agree that this is an obvious candidate for a refactor in the future.